eventhubs/client/azure-eventhubs/src/samples/java/ReceiveEventsByBatch.java

```diff
@@ -0,0 +1,98 @@
+        String connectionString = "Endpoint={endpoint};SharedAccessKeyName={sharedAccessKeyName};SharedAccessKey={sharedAccessKey};EntityPath={eventHubPath}";
+
+        // Instantiate a client that will be used to call the service.
+        EventHubClient client = new EventHubClientBuilder()
+            .connectionString(connectionString)
+            .build();
+
+        // To create a consumer, we need to know what partition to connect to. We take the first partition id.
+        // .blockFirst() here is used to synchronously block until the first partition id is emitted. The maximum wait
+        // time is set by passing in the OPERATION_TIMEOUT value. If no item is emitted before the timeout elapses, a
+        // TimeoutException is thrown.
+        String firstPartition = client.getPartitionIds().blockFirst(OPERATION_TIMEOUT);
+
+        // Create a consumer.
+        // The "$Default" consumer group is created by default. This value can be found by going to the Event Hub
+        // instance you are connecting to, and selecting the "Consumer groups" page. EventPosition.latest() tells the
+        // service we only want events that are sent to the partition after we begin listening.
+        EventHubConsumer consumer = client.createConsumer(EventHubClient.DEFAULT_CONSUMER_GROUP_NAME,
+            firstPartition, EventPosition.latest());
+
+        // We start receiving any events that come from `firstPartition`, print out the contents, and decrement the
+        // countDownLatch.
+        Disposable subscription = consumer.receive().subscribe(event -> {
+            String contents = UTF_8.decode(event.body()).toString();
+            System.out.println(String.format("[%s] Sequence Number: %s. Contents: %s", countDownLatch.getCount(),
+                event.sequenceNumber(), contents));
+
+            countDownLatch.countDown();
+        });
+
+        // Because the consumer is only listening to new events, we need to send some events to `firstPartition`.
+        // This creates a producer that only sends events to `firstPartition`.
+        EventHubProducerOptions producerOptions = new EventHubProducerOptions().partitionId(firstPartition);
+        EventHubProducer producer = client.createProducer(producerOptions);
+
+        // Create an event data list
+        ArrayList<EventData> events = new ArrayList<>(EVENT_BATCH_SIZE);
+        for (int i = 0; i < EVENT_BATCH_SIZE; i++) {
+            events.add(new EventData(UTF_8.encode("I am Event " + i)));
+        }
+
+        // We send a list of events to the service and block until the send has completed.
+        producer.send(events).block(OPERATION_TIMEOUT);
+        // We wait for all the events to be received before continuing.
+        countDownLatch.await(OPERATION_TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+
+        // Dispose and close of all the resources we've created.
+        subscription.dispose();
```

Comment:
nit:  Shouldn't these be in a `try/finally` block to prevent leaking in the face of exceptions?

Reply:
I thought we don't care about the exception from close() in the sample. It applies to all the samples we have.  But need double check with @conniey.

Reply:
It's possible that calling producer.send.block will throw an exception or countdown.await(), and even if those throw an exception,. you want to dispose of the subscription.

Reply:
Not sure that I agree.   If samples are to be the model for what we consider best practices,  I would think it important to show a structure where there are no leaks and things that need to be closed/disposed are.    In the worst case, I'd argue that we have an obligation to at least add a descriptive comment to that effect about how these should be closed and real-world code should expect exceptions and react appropriately.